### PR TITLE
RFC: Boot order redesign 

### PR DESCRIPTION
**Pull request by kostyanf14**

### CODE REVIEW

File: lib/setupmanagers/qemuhck/storage_manager.rb [hunk 1]
```diff
@@ -83,18 +84,32 @@ def boot_device_command(device_name, run_opts, qemu_replacement_map)
...
+
+        run_opts[:attach_iso_list]&.map do |iso|
+          iso_path = Pathname.new(@iso_path).join(iso)
+          boot_index += 1
+
+          "-device ide-cd,drive=iso_drive_#{boot_index},bus=ide.#{boot_index},bootindex=#{boot_index} " \
+            "-drive file=#{iso_path},if=none,media=cdrom,readonly=on,id=iso_drive_#{boot_index}"
+        end
+      end
+
```
Comment (akihikodaki) [inline lib/setupmanagers/qemuhck/storage_manager.rb:105]: Why would we want `bootindex` for ISOs?

File: lib/setupmanagers/qemuhck/qemu_machine.rb [hunk 2]
```diff
@@ -504,7 +507,7 @@ def base_cmd
       [
         '@qemu_bin@ -enable-kvm -machine @machine_options@ ',
         '-m @memory@,maxmem=@max_memory@ -smp @cpu_count@,cores=@cpu_count@ ',
-        '-cpu @cpu_options@ -boot order=cd,menu=on ',
+        '-cpu @cpu_options@ -boot menu=on,splash-time=10000 ',
```
Comment (akihikodaki) [inline lib/setupmanagers/qemuhck/qemu_machine.rb:510]: Please add a comment to describe the situation where `splash-time` is needed.

**Review by akihikodaki — CHANGES_REQUESTED** [review]

File: lib/setupmanagers/qemuhck/storage_manager.rb (hunk 1 above)
Comment (akihikodaki) [reply lib/setupmanagers/qemuhck/storage_manager.rb:105]: It's better to specify `bootindex` only for the install disk though I don't require to make a change for that.

Comment (kostyanf14) [reply lib/setupmanagers/qemuhck/storage_manager.rb:105]: Yes, this is a better solution,  but storage_manager and qemuhck do not know what image is really bootable.

Comment (akihikodaki) [reply lib/setupmanagers/qemuhck/storage_manager.rb:105]: A simple approach is to introduce a new option like `attach_installer_iso`.

Comment (kostyanf14) [reply lib/setupmanagers/qemuhck/storage_manager.rb:105]: Will move to separate task

File: lib/setupmanagers/qemuhck/qemu_machine.rb (hunk 2 above)
Comment (kostyanf14) [reply lib/setupmanagers/qemuhck/qemu_machine.rb:514]: done

**Review by akihikodaki — APPROVED** [review]

**Pull request merged**